core/server/worker/src/main/java/alluxio/worker/netty/DataServerShortCircuitWriteHandler.java

```diff
@@ -74,6 +77,13 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable throwable) {
     ctx.close();
   }
 
+  @Override
+  public void channelUnregistered(ChannelHandlerContext ctx) {
+    if (mSessionId > 0) {
+      mBlockWorker.cleanupSession(mSessionId);
+    }
```

Comment:
done. Because it is possible that mSessionId is not set. I think the current cleanSession code should also work if i do not have this check.